serf/messages_test.go

```diff
@@ -23,7 +23,7 @@ func TestQueryFlags(t *testing.T) {
 
 func TestEncodeMessage(t *testing.T) {
 	in := &messageLeave{Node: "foo"}
-	raw, err := encodeMessage(messageLeaveType, in)
+	raw, err := encodeMessage(messageLeaveType, in, false)
```

Comment:
Maybe we can add some tests here that encode/decode with both options a message that have a time field in it 🤔. Not sure, as I guess all of this is already tested in the library. WYT?

Reply:
I test this pretty thoroughly in the library https://github.com/hashicorp/go-msgpack/blob/main/codec/raft_test.go and in the Raft library https://github.com/hashicorp/raft/blob/main/util_test.go#L15